contracts/migrator/UniMigrator.sol

```diff
@@ -0,0 +1,175 @@
+            stakingToken,
+            amountSwapped,
+            amountAgEURMin,
+            amountTokenMin,
+            address(this)
+        );
+        if (gaugeType == 1) {
+            // In this case, it's _USDC
+            _UNI.createAndInitializePoolIfNecessary(_AGEUR, _USDC, 100, sqrtPriceX96Existing);
+            poolCreated = _GUNIFACTORY.createManagedPool(_AGEUR, _USDC, 100, 0, -276320, -273470);
+        } else {
+            // In this other case it's wETH
+            address pool = _UNI.createAndInitializePoolIfNecessary(_AGEUR, _WETH, 500, sqrtPriceX96Existing);
+            // Increasing observation cardinality on the new pool
+            IUniswapV3Pool(pool).increaseObservationCardinalityNext(144);
+            poolCreated = _GUNIFACTORY.createManagedPool(_AGEUR, _WETH, 500, 0, -96120, -69000);
+        }
+        console.log("poolCreated", poolCreated);
+        console.log("agEUR removed", amountAgEUR);
+        console.log("Token removed", amountToken);
+        IGUniPool(poolCreated).transferOwnership(0xe02F8E39b8cFA7d3b62307E46077669010883459);
+        uint256 newGUNIBalance;
+        (amountAgEUR, amountToken, newGUNIBalance) = _GUNIROUTER.addLiquidity(
+            poolCreated,
+            amountAgEUR,
+            amountToken,
+            amountAgEURMin,
+            amountTokenMin,
+            liquidityGauge
+        );
+        console.log("agEUR added", amountAgEUR);
+        console.log("Token added", amountToken);
+        console.log("Variation in GUNI positions");
+        console.log(amountSwapped, newGUNIBalance);
+        console.log("Scaling factor");
+        console.log((amountSwapped * 10**18) / newGUNIBalance);
+        ILiquidityGauge(liquidityGauge).set_staking_token_and_scaling_factor(
+            poolCreated,
+            (amountSwapped * 10**18) / newGUNIBalance
+        );
+        if (gaugeType == 1) {
+            ILiquidityGauge(liquidityGauge).commit_transfer_ownership(_GOVERNOR);
+            IERC20(_USDC).safeTransfer(_GOVERNOR, IERC20(_USDC).balanceOf(address(this)));
+        } else {
+            ethGUNIPool = poolCreated;
+            IERC20(_WETH).safeTransfer(_GOVERNOR, IERC20(_WETH).balanceOf(address(this)));
+        }
+        IERC20(_AGEUR).safeTransfer(_GOVERNOR, IERC20(_AGEUR).balanceOf(address(this)));
+    }
+
+    function finishPoolMigration(uint256 amountAgEURMin, uint256 amountETHMin) external onlyOwner {
+        uint256 amountRecoverable = IERC20(_GUNIETH).balanceOf(_ETHGAUGE);
+        ILiquidityGauge(_ETHGAUGE).recover_erc20(_GUNIETH, address(this), amountRecoverable);
+        uint256 amountAgEUR;
+        uint256 amountToken;
+        (amountAgEUR, amountToken, ) = _GUNIROUTER.removeLiquidity(
+            _GUNIETH,
+            amountRecoverable,
+            amountAgEURMin,
+            amountETHMin,
+            address(this)
+        );
+        console.log("Balances");
+        console.log(amountAgEUR, amountToken);
```

Comment:
In my tests it is respected. Normally scale factor for an amount of liquidity is defined by the first swap you're making.

Reply:
But let's say you have 20 g-uni, first half (10 g-UNI) can give you 9 new g-UNI, you have 10 old g-uni old, question is what happens if from the 10 old g-uni liquidity we don't get 9 new g-uni and if it's a problem

Reply:
Sounds like it would be a problem, if we get less than 9 new g-uni for the second half then the scaling factor would be wrong
Maybe we can recompute the scaling factor and re-update it in the gauge after the second half migration ?

Reply:
Otherwise, what about moving 99.99% of the gauge in the first part and then doing the remaining 0.01% ? Would that screw Euler ?